csrc/abstract_tensor.h

```diff
@@ -613,6 +613,10 @@ struct AbstractTensorWithInfo {
   bool operator==(T&& t) const {
     if constexpr (std::is_same_v<AbstractTensorWithInfo, std::decay_t<T>>) {
       return domain == t.domain && info == t.info;
+    } else if constexpr (std::is_same_v<
+                             std::vector<AbstractId>,
+                             std::decay_t<T>>) {
+      return domain == t;
```

Comment:
This fixes a failure in the `Reorder` test where we were no longer automatically casting `vector<AbstractId>` to `AbstractTensorWithInfo` so this returned false even with a matching domain provided.

Reply:
To avoid this kind of silent error again, should we just remove the template `T` here and implement this only for `AbstractTensorWithInfo` and `std::vector<AbstractId>`?

Reply:
Makes sense to me.